File: stack.cpp
#include <bits/stdc++.h>
using namespace std;

class Stack
{
public:
    int top;
    vector<int> s;

    Stack()
    {
        top = -1;
    }
    void isEmpty()
    {
        if (top == -1)
        {
            cout << "Stack is empty";
        }
        cout << "No";
    }
    void push(int x)
    {
        top++;
        s.push_back(x);
    }

    void pop()
    {
        s.pop_back();
        top--;
    }

    void viewtop()
    {
        cout << "Top of the stack: " << s[top] << endl;
    }
    void traverse()
    {
        cout << "Stack: ";
        for (int i = 0; i < s.size(); i++)
        {
            cout << s[i] << " ";
        }
        cout << endl;
    }
};

int main()
{
    Stack arr;
    arr.traverse();
    arr.push(1);
    arr.push(2);
    arr.push(3);
    arr.push(4);
    arr.traverse();
    arr.viewtop();
    arr.pop();
    arr.pop();
    arr.viewtop();
    arr.traverse();
}

File: stack.py
class Stack:
    def __init__(self, cap):
        self.stack = []
        self.capacity = cap

    # Time Complexity: O(1)
    def push(self, value):
        if len(self.stack) == self.capacity:
            print("Stack is full")
        else:
            self.stack.append(value)

    # Time Complexity: O(1)
    def pop(self):
        if not len(self.stack):
            print("Stack is Empty")
        else:
            self.stack.pop(-1)

    # Time Complexity: O(1)
    def isEmpty(self):
        return not len(self.stack)

    # Time Complexity: O(1)
    def isFull(self):
        return len(self.stack) == self.capacity

    # Time Complexity: O(1)
    def peek(self):
        if not len(self.stack):
            print("Stack is Empty")
        else:
            print(self.stack[-1])


newStack = Stack(5)
print(newStack.stack)
